### PR TITLE
Allow compilation of tests as an external project 

### PR DESCRIPTION
**Pull request by GiulioRomualdi**
With this PR:
1. move `OsqpEigen.h` into `include/` folder
2. implement the possibility to compile the `tests` folder outside the `osqp-eigen` project
3. remove `OSQPWrapper` leftovers in `tests`

For further details check #27

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -69,7 +69,7 @@ set(${LIBRARY_TARGET_NAME}_SRC
   src/Solver.cpp)
 
 set(${LIBRARY_TARGET_NAME}_HDR
-  include/OsqpEigen.h
+  include/OsqpEigen/OsqpEigen.h
```
Comment (S-Dafarra) [inline CMakeLists.txt]: Isn't this breaking the compatibility with other projects?

Comment (S-Dafarra) [reply CMakeLists.txt]: Probably not, as we were already doing ``#include <OsqpEigen/OsqpEigen.h>`` as in https://github.com/robotology/walking-controllers/blob/master/modules/Walking_module/include/WalkingQPInverseKinematics_osqp.hpp

**Comment by S-Dafarra** [comment]
If we can just check that this change won't break the projects that use ``osqp-eigen``, then it is good for me.

File: CMakeLists.txt (hunk 1 above)
Comment (traversaro) [reply CMakeLists.txt]: The installation layout was already `<install_prefix>/OsqpEigen/OsqpEigen.h`, this just aligns the source layout with the  install layout.

**Comment by GiulioRomualdi** [comment]
Tested, everything seems ok

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -5,37 +5,44 @@ cmake_minimum_required(VERSION 3.1)
...
+  include_directories(SYSTEM ${EIGEN3_INCLUDE_DIR})
+endif()
 
 find_package(GTest REQUIRED)
 include_directories(${GTEST_INCLUDE_DIRS})
 
 # QPTest
 add_executable(SparseMatrixTest SparseMatrixTest.cpp)
-target_link_libraries(SparseMatrixTest OsqpEigen osqp::osqp ${GTEST_LIBRARIES} pthread)
+target_link_libraries(SparseMatrixTest OsqpEigen::OsqpEigen osqp::osqp Eigen3::Eigen ${GTEST_LIBRARIES} pthread)
```
Comment (traversaro) [inline tests/CMakeLists.txt]: Given that both `osqp` and `Eigen3` headers are included in`OsqpEigen`'s  public headers, if you turn the line `target_link_libraries(${LIBRARY_TARGET_NAME} PRIVATE osqp::osqp Eigen3::Eigen)` in `target_link_libraries(${LIBRARY_TARGET_NAME} PUBLIC osqp::osqp Eigen3::Eigen)` the main CMakeLists.txt, you do not need  anymore to manually link `osqp::osqp` and  `Eigen3::Eigen`. However, I know realized the big problem of `FindEigen3.cmake` etc etc, so I understand why we do not do that.

**Pull request merged**